common/src/main/java/org/apache/uniffle/common/security/HadoopSecurityContext.java

```diff
@@ -38,6 +40,7 @@ public class HadoopSecurityContext implements SecurityContext {
 
   private UserGroupInformation loginUgi;
   private ScheduledExecutorService refreshScheduledExecutor;
+  private Map<String, UserGroupInformation> proxyUserUgiPool;
```

Comment:
Could we use a LRU cache instead of map?

Reply:
Not necessary, the number of hadoop user is limited.

Reply:
> Not necessary, the number of hadoop user is limited.

What's the maxValue?

Reply:
if use LRU, the memory leak problem still exists. By the way, should we use JavaUtils here?

Reply:
> if use LRU, the memory leak problem still exists.

Right answer! +1

Reply:
Could we add some comments to explain this?